test/unit/accord/utils/random/Picker.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accord.utils.random;
+
+import java.util.Arrays;
+import java.util.function.Supplier;
+
+import accord.utils.Invariants;
+import accord.utils.RandomSource;
+
+public class Picker
+{
+    public static float[] randomWeights(RandomSource random, int length)
+    {
+        float[] weights = new float[length - 1];
+        float sum = 0;
+        for (int i = 0 ; i < weights.length ; ++i)
+            weights[i] = sum += random.nextFloat();
+        sum += random.nextFloat();
+        for (int i = 0 ; i < weights.length ; ++i)
+            weights[i] /= sum;
+        return weights;
+    }
+
+    static abstract class Weighted
+    {
+        final RandomSource random;
+        final float[] weights;
+
+        public Weighted(RandomSource random, float[] weights)
+        {
+            this.random = random;
+            this.weights = weights;
+        }
+
+
+        static float[] randomWeights(RandomSource random, float[] bias)
+        {
+            float[] weights = new float[bias.length - 1];
```

Comment:
Do we ignore one of the `bias` floats?

Reply:
its very leaky... you must use it "correctly" and only once place in the code knows what that means...